test/e2e/proxy_test.go

```diff
@@ -32,6 +34,44 @@ func proxyTestSuite(t *testing.T) {
 
 	t.Run("Trusted CA ConfigMap validation", testTrustedCAConfigMap)
 	t.Run("Environment variables validation", testEnvVars)
+	t.Run("Certificate validation", testCerts)
+}
+
+func testCerts(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
```

Comment:
I missed this in previous proxy PRs, but this pattern of creating a new testContext within each individual test has been removed from the rest of the e2e tests. Instead NewTestContext is called once at the beginning of each test suite, and the tests which require its use are made methods of it.

The reason for this ask is that by using the previous pattern, each test ends up having a repeated boiler plate code within them. 

If you can please add a commit changing this for the rest of the tests. If you rather not do that within this PR I am fine with it being done only for this test. Meaning within this PR changing only this function to be a method of testContext, without changing the others.